### PR TITLE
Fix java code generation

### PR DESCRIPTION
**Pull request by tbetbetbe**
- Updates packager and api_repo.js to handle java packages

- Corrects java package generation to so that the generated java package
  contains a functioning build.gradle the generates a java package
  containing the protos.

### CODE REVIEW

**Comment by tbetbetbe** [comment]
FYI: @ejona86, @garrettjonesgoogle

File: lib/api_repo.js [hunk 1]
```diff
@@ -176,6 +179,7 @@ ApiRepo.prototype.buildPackages =
           opts.packageInfo.api.simplename = name;
           opts.packageInfo.api.name = that.pkgPrefix + name;
           opts.packageInfo.api.version = version;
+          console.log('calling packager.%s', l);
```
Comment (saicheems) [inline lib/api_repo.js]: ?

File: lib/api_repo.js [hunk 2]
```diff
@@ -309,7 +316,22 @@ ApiRepo.prototype._buildProtos =
...
       finder.startSearch();
     };
-    if (language === 'nodejs') {
+    if (language === 'java') {
+      var dstTopDir = path.join(that.outDir, language);
+      var dstResourceDir = path.join(dstTopDir, 'src', 'main', 'resources');
+      fs.mkdirsSync(dstResourceDir);
+      /**
+       * copyJavaPb copies a protocol buffer file to the java package resource
+       * folder
```
Comment (saicheems) [inline lib/api_repo.js]: end with period

File: lib/api_repo.js [hunk 3]
```diff
@@ -309,7 +316,22 @@ ApiRepo.prototype._buildProtos =
...
+       */
+      var copyJavaPb = function copyJavaPb(protoPath, next) {
+        var src = path.join(that.repoDir, protoPath),
+            dst = path.join(dstResourceDir, protoPath);
+        console.log('Copying %s to %s', src, dst);
```
Comment (saicheems) [inline lib/api_repo.js]: intended?

File: lib/packager.js [hunk 4]
```diff
@@ -195,6 +209,49 @@ function makePythonPackage(opts, done) {
...
+    if (dstBase === 'build.gradle.mustache') {
+      dstBase = 'build.gradle';
+    }
+    var tmpl = path.join(opts.templateDir, f)
+      , dst = path.join(opts.top, dstBase);
+    tasks.push(expand.bind(null, tmpl, dst, opts.packageInfo));
+  });
+
+  async.parallel(tasks, function(err) {
+    console.log('what is err %s', err);
```
Comment (saicheems) [inline lib/packager.js]: ?

File: lib/packager.js [hunk 5]
```diff
@@ -195,6 +209,49 @@ function makePythonPackage(opts, done) {
...
+ */
+function makeJavaPackage(opts, done) {
+  opts = _.merge({}, settings.java, opts);
+  var tasks = [];
+
+  // Move copyable files to the top-level dir.
+  opts.copyables.forEach(function(f) {
+    var src = path.join(opts.templateDir, f)
+      , dst = path.join(opts.top, f);
+    console.log('Copying %s to %s', src, dst);
```
Comment (saicheems) [inline lib/packager.js]: intended?

**Comment by saicheems** [comment]
lgtm

File: lib/api_repo.js (hunk 1 above)
Comment (tbetbetbe) [inline lib/api_repo.js]: Fixed

File: lib/api_repo.js (hunk 2 above)
Comment (tbetbetbe) [inline lib/api_repo.js]: Fixed

File: lib/api_repo.js (hunk 3 above)
Comment (tbetbetbe) [inline lib/api_repo.js]: Fixed

File: lib/packager.js (hunk 5 above)
Comment (tbetbetbe) [inline lib/packager.js]: Fixed

**Pull request merged**